libraries/AP_ADSB/AP_ADSB.cpp

```diff
@@ -166,51 +168,73 @@ const AP_Param::GroupInfo AP_ADSB::var_info[] = {
 AP_ADSB::AP_ADSB()
 {
     AP_Param::setup_object_defaults(this, var_info);
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
     if (_singleton != nullptr) {
         AP_HAL::panic("AP_ADSB must be singleton");
     }
+#endif
     _singleton = this;
+
+#ifdef ADSB_STATIC_CALLSIGN
+    memcpy(&out_state.cfg.callsign, ADSB_STATIC_CALLSIGN, 8);
```

Comment:
Should be using sizeof-the-target.

Further, what guarantees on the length of `ADSB_STATIC_CALLSIGN` are there?  Would someone defining the symbol know they exist?

Reply:
changed to:

`#ifdef ADSB_STATIC_CALLSIGN`
    `const uint8_t callsign_len = MIN(sizeof(out_state.cfg.callsign),sizeof(ADSB_STATIC_CALLSIGN));`
    `memcpy(&out_state.cfg.callsign, ADSB_STATIC_CALLSIGN, callsign_len);`
`#endif`